### DISCUSSION TITLE
Why do editor addons run on project pages

### DISCUSSION THREAD

**Discussion by Samq64** [ScratchAddons/ScratchAddons#5059] [Help] [answered]
Why exactly do all editor addons run on the project page? I know switching between the project page and editor don't reload the page, but that shouldn't be a problem, right? Are the editor and project pages just hidden with `display: none;` when switched between them?

**Comment by mxmou** [comment, marked as answer]
They have to run on page load to be able to access the VM.

**Reply by Samq64** [reply]
What about addons that don't use the VM?

**Reply by WorldLanguages** [reply]
Userscript URL matching happens with the initial URL. Going inside changes the URL but it's not actually a different page.

We could make it so editor addons don't get injected until the user goes inside for the first time, tho. I just think we never considered it.

**Comment by CST1229** [comment]
I'm pretty sure the player and editor aren't just switched between with display: none.

**Reply by WorldLanguages** [reply]
When going inside or outside, Scratch destroys the whole DOM and builds it again (well, React actually).
So every time the user goes inside or outside, addons need to make their respective changes to the UI again.
Editor addons that don't have an effect on the project page just don't do anything when the user goes outside, it's removed by Scratch already
